ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/views/imagebrowser/repositories/AbstractRepository.java

```diff
@@ -177,7 +186,10 @@ protected void searchDirectory(File directory, final IProgressMonitor monitor) {
 
 						} else {
 							if (isImage(resource)) {
-								addImageElement(new ImageElement(() -> createImageData(resource), pluginName, resource.getAbsolutePath().substring(directoryPathLength)));
+								ImageElement imageElement = new ImageElement(() -> createImageData(resource),
+										pluginName, resource.getAbsolutePath().substring(directoryPathLength));
+								mElementsCache.put(imageElement.getPath(), imageElement);
+								addImageElement(imageElement);
```

Comment:
Can't this code be left unchanged?
```suggestion
								addImageElement(new ImageElement(() -> createImageData(resource), pluginName, resource.getAbsolutePath().substring(directoryPathLength)));
```